### PR TITLE
SAK-33603 - Assignment Soft Deletion UI

### PR DESCRIPTION
**Pull request by mpellicer**
0c682c0ae880be39b417d2d26807f837 1

### CODE REVIEW

**Comment by mpellicer** [comment]
This has a design flaw, if there are no assignments the menu is not accessible. I'm moving the option to the top bar so please don't merge this.

**Comment by mpellicer** [comment]
Option moved, now it looks like this:
image

**Comment by ottenhoff** [comment]
LGTM

File: assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java [hunk 1]
```diff
@@ -384,6 +384,15 @@
...
     public Collection<Assignment> getAssignmentsForContext(String context);
 
+    /**
+     * Access all the Assignments that are deleted
+     *
+     * @param context -
+     *                Describes the portlet context - generated with DefaultId.getChannel().
+     * @return List All the deleted assignments will be listed
+     */
+    public Collection<Assignment> getDeletedAssignmentsForContext(String context);
```
Comment (ern) [inline assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java]: Lets add a unit test for this method...

Comment (mpellicer) [reply assignment/api/src/java/org/sakaiproject/assignment/api/AssignmentService.java]: You're right @ern, I've added a unit test, it deletes an assignment and checks the returned Collection.

**Pull request merged**